src/procedures/proc_procedures.c

```
@@ -15,111 +15,123 @@ extern rax *__procedures;
-	ProcGenerator gen = pdata->iter.data;
-	ProcedureCtx *curr_proc_ctx = gen();
-
-	if(pdata->yield_name) *pdata->yield_name =
-			SI_ConstStringVal(Procedure_GetName(curr_proc_ctx));
-	if(pdata->yield_mode) *pdata->yield_mode =
-			Procedure_IsReadOnly(curr_proc_ctx) ? SI_ConstStringVal("READ") :
-			SI_ConstStringVal("WRITE");
-
-	Proc_Free(curr_proc_ctx);
-	return pdata->output;
+    ASSERT(ctx->privateData);
```

cppcheck

[error] Null pointer dereference

(nullPointer)